### PR TITLE
Certificates metrics monitoring

### PR DESCRIPTION
**Pull request by sahma19**
This is to be released with changes in https://github.com/Energinet-DataHub/eo-base-environment/pull/473

### CODE REVIEW

File: domains/certificates/Query.API/API/API.csproj [hunk 1]
```diff
@@ -17,6 +17,13 @@
     <PackageReference Include="MassTransit.RabbitMQ" Version="8.0.12" />
     <PackageReference Include="Microsoft.AspNetCore.Authentication.JwtBearer" Version="7.0.4" />
     <PackageReference Include="Microsoft.Extensions.Http.Polly" Version="7.0.4" />
+    <PackageReference Include="OpenTelemetry.Exporter.OpenTelemetryProtocol" Version="1.5.0-alpha.2" />
```
Comment (duizer) [inline domains/certificates/Query.API/API/API.csproj:20]: Do we use this?

File: domains/certificates/Query.API/API/API.csproj [hunk 2]
```diff
@@ -17,6 +17,13 @@
...
+    <PackageReference Include="OpenTelemetry.Exporter.Prometheus.AspNetCore" Version="1.5.0-alpha.1" />
+    <PackageReference Include="OpenTelemetry.Extensions.Hosting" Version="1.5.0-alpha.2" />
+    <PackageReference Include="OpenTelemetry.Instrumentation.AspNetCore" Version="1.0.0-rc9.14" />
+    <PackageReference Include="OpenTelemetry.Instrumentation.Http" Version="1.0.0-rc9.14" />
+    <PackageReference Include="OpenTelemetry.Instrumentation.Process" Version="0.5.0-beta.2" />
```
Comment (duizer) [inline domains/certificates/Query.API/API/API.csproj:25]: I do not think this is used in Program.cs, but I think we should be using it.

File: domains/certificates/Query.API/API/API.csproj (hunk 1 above)
Comment (duizer) [reply domains/certificates/Query.API/API/API.csproj]: No we do not. It is removed now

File: domains/certificates/Query.API/API/API.csproj (hunk 2 above)
Comment (duizer) [reply domains/certificates/Query.API/API/API.csproj:24]: It is being used now

File: domains/certificates/chart/templates/api-service.yaml [hunk 3]
```diff
@@ -3,6 +3,8 @@ kind: Service
 metadata:
   name: {{ .Release.Name }}-api
   namespace: {{ .Release.Namespace }}
+  labels:
+    app: {{ .Release.Name }}-api
```
[the remark this answers is not in the document]
Comment (duizer) [reply domains/certificates/chart/templates/api-service.yaml:7]: Service added if prometheus support is enabled

File: domains/certificates/chart/values.yaml [hunk 4]
```diff
@@ -12,3 +12,7 @@ postgres:
   size: 10Gi
 rabbitmq:
   replicaCount: 1
+metrics:
+  prometheus:
+    enabled: true
+    # serviceMonitorReleaseLabel: ""
```
Comment (CodeReaper) [inline domains/certificates/chart/values.yaml:18]: I would like to have `additionalLabels` where you can add both key and value for labels. It is much more open for configuration and less confusing about what a "release" label is.

File: domains/certificates/Query.API/API/API.csproj [hunk 5]
```diff
@@ -17,6 +17,12 @@
     <PackageReference Include="MassTransit.RabbitMQ" Version="8.0.12" />
     <PackageReference Include="Microsoft.AspNetCore.Authentication.JwtBearer" Version="7.0.4" />
     <PackageReference Include="Microsoft.Extensions.Http.Polly" Version="7.0.4" />
+    <PackageReference Include="OpenTelemetry.Exporter.Prometheus.AspNetCore" Version="1.5.0-alpha.1" />
+    <PackageReference Include="OpenTelemetry.Extensions.Hosting" Version="1.5.0-alpha.2" />
+    <PackageReference Include="OpenTelemetry.Instrumentation.AspNetCore" Version="1.0.0-rc9.14" />
+    <PackageReference Include="OpenTelemetry.Instrumentation.Http" Version="1.0.0-rc9.14" />
+    <PackageReference Include="OpenTelemetry.Instrumentation.Process" Version="0.5.0-beta.2" />
+    <PackageReference Include="OpenTelemetry.Instrumentation.Runtime" Version="1.1.0-rc.2" />
```
Comment (CodeReaper) [inline domains/certificates/Query.API/API/API.csproj:25]: Are there not any _released_ versions that we could use instead?

Comment (duizer) [reply domains/certificates/Query.API/API/API.csproj:25]: Unfortunately not: https://github.com/open-telemetry/opentelemetry-dotnet/releases. It is the same case for Auth-domain

File: domains/certificates/chart/values.yaml (hunk 4 above)
Comment (duizer) [reply domains/certificates/chart/values.yaml:18]: So which objects should these labels be added to? In certificates domain we have Deployment-objects, Service-objects, ConfigMap-objects, StatefulSet-objects, RabbitMQCluster-objects and ServiceMonitor-objects. Should the labels in `additionalLabels` be added to all of these objects?

Comment (duizer) [reply domains/certificates/chart/values.yaml]: I have added `additionalLabels` in 24c11053364fbd104da0ea4ecc832a9d47000dad, but it only adds labels to the ServiceMonitor. Please let me know, if this is what you are asking for.

**Review by duizer — APPROVED** [review]

**Comment by ckr123** [comment]
does this include a grafana board?

**Review by ckr123 — DISMISSED** [review]

File: domains/certificates/chart/templates/api-service.yaml (hunk 3 above)
Comment (duizer) [inline domains/certificates/chart/templates/api-service.yaml:7]:
```suggestion
    service: {{ .Release.Name }}-api
```

File: domains/certificates/chart/templates/service-monitor.yaml [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    {{- end }}
+  {{- end }}
+spec:
+  endpoints:
+    - port: http
+      interval: 15s
+      scrapeTimeout: 14s
+  selector:
+    matchExpressions:
+      - key: app
```
Comment (duizer) [inline domains/certificates/chart/templates/service-monitor.yaml:19]:
```suggestion
      - key: service
```

File: domains/certificates/chart/templates/registry-connector-service.yaml [hunk 7]
```diff
@@ -0,0 +1,18 @@
+{{- if .Values.metrics.prometheus.enabled -}}
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{ .Release.Name }}-registry-connector
+  namespace: {{ .Release.Namespace }}
+  labels:
+    app: {{ .Release.Name }}-registry-connector
```
Comment (duizer) [inline domains/certificates/chart/templates/registry-connector-service.yaml:8]:
```suggestion
    service: {{ .Release.Name }}-registry-connector
```

**Review by CodeReaper — APPROVED** [review]

**Pull request merged**